packages/rum-core/src/domain/startUserDataTelemetry.ts

```diff
@@ -0,0 +1,119 @@
+import type { BatchFlushEvent, Context, ContextManager, Observable, Telemetry } from '@datadog/browser-core'
+import {
+  isExperimentalFeatureEnabled,
+  performDraw,
+  ONE_SECOND,
+  addTelemetryDebug,
+  monitor,
+} from '@datadog/browser-core'
+import { RumEventType } from '../rawRumEvent.types'
+import type { RumEvent } from '../rumEvent.types'
+import type { RumConfiguration } from './configuration'
+import type { FeatureFlagContexts } from './contexts/featureFlagContext'
+import type { LifeCycle } from './lifeCycle'
+import { LifeCycleEventType } from './lifeCycle'
+
+export const MEASURES_FLUSH_INTERVAL = 10 * ONE_SECOND
+
+type Measure = {
+  min: number
+  max: number
+  sum: number
+}
+
+let batchCount: number
+let batchBytesCount: Measure
+let batchMessagesCount: Measure
+let globalContextBytes: Measure
+let userContextBytes: Measure
+let featureFlagBytes: Measure
+
+let currentBatchGlobalContextBytes: Measure
+let currentBatchUserContextBytes: Measure
+let currentBatchFeatureFlagBytes: Measure
+
+export function startUserDataTelemetry(
+  configuration: RumConfiguration,
+  telemetry: Telemetry,
+  lifeCycle: LifeCycle,
+  globalContextManager: ContextManager,
+  userContextManager: ContextManager,
+  featureFlagContexts: FeatureFlagContexts,
+  batchFlushObservable: Observable<BatchFlushEvent>
+) {
+  const userDataTelemetryEnabled = telemetry.enabled && performDraw(configuration.userDataTelemetrySampleRate)
+  if (!userDataTelemetryEnabled || !isExperimentalFeatureEnabled('user_data_telemetry')) {
+    return
+  }
+
+  initMeasures()
+  initCurrentBatchMeasures()
+
+  lifeCycle.subscribe(LifeCycleEventType.RUM_EVENT_COLLECTED, (event: RumEvent & Context) => {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏it could be helpful to add a comment to warn that measures here are updated for each view updates.
It should not be problematic now but it could be easily missed if we need to collect more stuff.

Reply:
I don't really get your comment

Reply:
I added this [comment](https://github.com/DataDog/browser-sdk/blob/81d2864f77d6b8c86af72a166587b85c3de2787b/packages/rum-core/src/domain/startCustomerDataTelemetry.ts#L62). Let me know what you think